src/DGtal/geometry/tools/QuickHullKernels.h

```diff
@@ -0,0 +1,805 @@
+    // inheriting constants
+    using Base::dimension;
+    // inheriting methods
+    using Base::compute;
+    using Base::normal;
+    using Base::intercept;
+    using Base::dot;
+    using Base::equal;
+    using Base::height;
+    using Base::volume;
+    using Base::above;
+    using Base::aboveOrOn;
+    using Base::on;
+
+    /// The precision as the common denominator for all rational points.
+    double precision;
+    
+    /// Constructor with specified precision
+    ///
+    /// @param[in] aPrecision the chosen precision as the common
+    /// denominator of all rationals (by defaut, 1024).
+    ConvexHullRationalKernel( double aPrecision = 1024. )
+      : precision( aPrecision ) {}
+
+    /// @return 'true' if a kernel may induce infinite facets. This is
+    /// typically the case of a Delaunay computation kernel, which
+    /// casts points in higher dimension.
+    bool hasInfiniteFacets() const
+    { return false; }
+
+    /// @param[in] hs an half-space corresponding to a facet.
+    ///
+    /// @return 'true' if the facet associated to this half-space
+    /// corresponds to an infinite facet.
+    bool isHalfSpaceFacetInfinite( const HalfSpace& hs ) const
+    {
+      (void) hs; // unused parameter      
+      return false;
+    }
+    
+    /// Transforms a range \a input_points of input points to a range
+    /// \a processed_points of points adapted to a processing by
+    /// QuickHull convex hull algorithm. Keep the mapping information
+    /// between input and processed points.
+    ///
+    /// @tparam InputPoint any model of point whose components are
+    /// convertible to Scalar.
+    ///
+    /// @param[out] processed_points the range of points prepared for
+    /// a process by QuickHull.
+    /// 
+    /// @param[out] input2comp the surjective mapping between
+    /// the \a input_points range and the \a processed_points range
+    /// used for computation.
+    ///
+    /// @param[out] comp2input the injective mapping between the 
+    /// \a processed_points range used for computation and the \a
+    /// input_points range.
+    ///
+    /// @param[in] input_points the range of input points.
+    ///
+    /// @param[in] remove_duplicates when 'true', this method removes possible
+    /// duplicates in \a input_points and \a processed_points may thus
+    /// be of smaller size, otherwise, when 'false', it means that
+    /// there are no duplicates in \a input_points.
+    template < typename InputPoint>
+    void makeInput( std::vector< Point >& processed_points,
+                    IndexRange& input2comp, IndexRange& comp2input,
+                    const std::vector< InputPoint >& input_points,
+                    bool remove_duplicates )
+    {
+      const auto F = [&] ( InputPoint input ) -> Point
+        {
+          Point p;
+          for ( Dimension i = 0; i < dimension; i++ )
+            p[ i ] = static_cast< Scalar >( round( input[ i ] * precision ) );
```

Comment:
This conversion formula should  probably be more explicit in the class/function description (relying on rounding on doubles)

Reply:
Doc added here and in the kernel class to make more explicit the conversions.